pkg/kubectl/cmd/resource/get.go

```diff
@@ -296,8 +310,8 @@ func (options *GetOptions) Run(f cmdutil.Factory, cmd *cobra.Command, args []str
 		filterFuncs = nil
 	}
 
-	if printer.IsGeneric() {
-		return options.printGeneric(printer, r, filterFuncs, filterOpts)
+	if isGeneric {
```

Comment:
All this ugly exists because of "human readable printers" that want to have special rules for column headers, right?

However, in that printer you know many of the conditions you're checking here since the order of PrintObject calls matters.  You'd be able to detect things like "should I print the column headers" based on checking against the previous `PrintObject` call.

The Printer for all human readable printers should be the same.  That's the thing you're realizing in this code.  The printer is always chosen during `Complete` and it needs to manage printing any Object.

Reply:
Yes, it's already pointed out in the TODO where this flag is defined.